rmgpy/molecule/resonance.py

```diff
@@ -341,17 +389,187 @@ def generate_isomorphic_isomers(mol):
+        bonds.update([atom.bonds[key] for key in atom.bonds.keys() if key.isNonHydrogen()])
+    bonds = list(bonds)
+
+    # Identify exocyclic bonds, and save their bond orders
+    exo = []
+    for bond in bonds:
+        if bond.atom1 not in atoms or bond.atom2 not in atoms:
+            if bond.order == 'D':
+                exo.append(1)
+            else:
+                exo.append(0)
+        else:
+            exo.append(None)
+
+    # Dimensions
+    l = len(ASSSR)
+    m = len(atoms)
+    n = l + len(bonds)
+
+    # Connectivity matrix which indicates which rings and bonds each atom is in
+    # Part of equality constraint Ax=b
+    a = []
+    for atom in atoms:
+        inRing = [1 if atom in ring else 0 for ring in ASSSR]
+        inBond = [1 if atom in [bond.atom1, bond.atom2] else 0 for bond in bonds]
+        a.append(inRing + inBond)
+
+    # Objective vector for optimization: sextets have a weight of 1, double bonds have a weight of 0
+    objective = [1] * l + [0] * len(bonds)
+
+    # Solve LP problem using lpsolve
+    lp = lpsolve('make_lp', m, n)               # initialize lp with constraint matrix with m rows and n columns
+    lpsolve('set_verbose', lp, 2)               # reduce messages from lpsolve
+    lpsolve('set_obj_fn', lp, objective)        # set objective function
+    lpsolve('set_maxim', lp)                    # set solver to maximize objective
+    lpsolve('set_mat', lp, a)                   # set left hand side to constraint matrix
+    lpsolve('set_rh_vec', lp, [1] * m)          # set right hand side to 1 for all constraints
+    lpsolve('set_constr_type', lp, ['='] * m)   # set all constraints as equality constraints
+    lpsolve('set_binary', lp, [True] * n)       # set all variables to be binary
+
+    # Constrain values of exocyclic bonds, since we don't want to modify them
+    for i in range(l, n):
+        if exo[i - l] is not None:
+            # NOTE: lpsolve indexes from 1, so the variable we're changing should be i + 1
+            lpsolve('set_bounds', lp, i + 1, exo[i - l], exo[i - l])
+
+    # Add constraints to problem if provided
+    if constraints is not None:
+        for constraint in constraints:
+            lpsolve('add_constraint', lp, constraint[0], '<=', constraint[1])
+
+    status = lpsolve('solve', lp)
+    objVal, solution = lpsolve('get_solution', lp)[0:2]
+    lpsolve('delete_lp', lp)  # Delete the LP problem to clear up memory
+
+    # Check that optimization was successful
+    if status != 0:
+        raise RuntimeError('Optimization did not obtain optimal solution')
+
+    # Check that we the result contains at least one aromatic sextet
+    if objVal == 0:
+        return []
+
+    # Check that the solution contains the maximum number of sextets possible
+    if maxNum is None:
+        maxNum = objVal  # This is the first solution, so the result should be an upper limit
+    elif objVal < maxNum:
+        raise ValueError('Sub-optimal solution obtained.')
```

Comment:
IMO (which is not necessarily that of a good programmer), it is okay to raise built-in error types if you don't ever expect to catch the error because the person running the code can always read the message and tell what happened. 

In this case, I think you should create a new error type like "InvalidILPSolutionError", so that you don't confuse it with actual Value Errors that could occur in other places of your code.